ksql-common/src/main/java/io/confluent/ksql/schema/ksql/types/Field.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql.types;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.util.StringUtil;
+import java.util.Objects;
+
+@Immutable
+public final class Field {
+
+  private final String name;
+  private final SqlType type;
+
+  public static Field of(final String name, final SqlType type) {
+    return new Field(name, type);
+  }
+
+  private Field(final String name, final SqlType type) {
+    this.name = Objects.requireNonNull(name, "name");
+    this.type = Objects.requireNonNull(type, "type");
```

Comment:
throw on empty / non-trimmed field name?

Reply:
Added unit tests for these so it isn't missed again!